packages/go_router/lib/src/route_data.dart

```diff
@@ -63,6 +63,11 @@ abstract class GoRouteData extends RouteData {
   /// Corresponds to [GoRoute.redirect].
   FutureOr<String?> redirect(BuildContext context, GoRouterState state) => null;
 
+  /// Called when this route is removed from GoRouter's route history.
+  ///
+  /// Corresponds to [GoRoute.onExit].
+  FutureOr<bool> onExit(BuildContext context) => true;
```

Comment:
is this from other patch? we should probably separate this out

Reply:
Oops yeah, that's a mistake. I reverted it